### PR TITLE
Added InlineAutocomplete stories

### PR DESCRIPTION
**Pull request by green6erry**
Parent issue: https://github.com/github/primer/issues/1488

Add storybook stories for InlineAutocomplete (draft)

Checklist

- [x]  Ensure your chosen component has a component directory in the src directory
- [x]  Ensure the component has a root story file in the component directory
- [x]  Ensure that the root story file exports a Default story
- [x]  (Optional) Create a Playground story
- [x]  Move all other stories for the component into a .features.stories.tsx file
- [x]  Opt the component into storybook tests
- [ ]  (Optional) Create visual regression tests for the component

### CODE REVIEW

**Comment by broccolinisoup** [comment]
Hello @green6erry would that be possible to resolve the merge conflicts before the review? Thank you 🙏🏻

**Comment by green6erry** [comment]
> Hello @green6erry would that be possible to resolve the merge conflicts before the review? Thank you 🙏🏻

Done!

**Review by broccolinisoup — CHANGES_REQUESTED** [review]
Hello 👋🏻 VRT tests are failing, because they are still running on the "old" story ids. 

Because we renamed component stories in this PR from being `Components/Forms/InlineAutocomplete/...` to `Components/InlineAutocomplete/..` , their ids are changed too and we need to update them on https://github.com/primer/react/blob/main/e2e/components/InlineAutocomplete.test.ts so that the Playwright can run on these stories. Let me know if you have any questions! 

We will need to update the Playwright

File: src/InlineAutocomplete/InlineAutocomplete.features.stories.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import React, {useState} from 'react'
 import {Meta} from '@storybook/react'
-import {ActionList, Avatar, ActionListItemProps, Textarea, TextInput, FormControl} from '../..'
+import {ActionList, Avatar, ActionListItemProps, Textarea, TextInput, FormControl} from '..'
 import InlineAutocomplete, {ShowSuggestionsEvent, Suggestions} from '.'
 
 export default {
-  title: 'Components/Forms/InlineAutocomplete/Features',
+  title: 'Components/InlineAutocomplete/Features',
```
Comment (broccolinisoup) [inline src/InlineAutocomplete/InlineAutocomplete.features.stories.tsx:7]: Sorry, I didn't realised this component is draft. For draft components, we should serve them under the Drafts folder in storybook. 
```suggestion
  title: 'Drafts/Components/InlineAutocomplete/Features',
```

File: src/InlineAutocomplete/InlineAutocomplete.stories.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import React, {useState} from 'react'
 import {Meta} from '@storybook/react'
 
-import {BaseStyles, Box, Textarea, ThemeProvider, FormControl} from '../..'
+import {BaseStyles, Box, Textarea, ThemeProvider, FormControl} from '..'
 import InlineAutocomplete, {ShowSuggestionsEvent, Suggestions} from '.'
 
 export default {
-  title: 'Components/Forms/InlineAutocomplete',
+  title: 'Components/InlineAutocomplete',
```
Comment (broccolinisoup) [inline src/InlineAutocomplete/InlineAutocomplete.stories.tsx:8]:
```suggestion
  title: 'Drafts/Components/InlineAutocomplete',
```

File: e2e/components/InlineAutocomplete.test.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ test.describe('InlineAutocomplete', () => {
       test.describe(theme, () => {
         test('default @vrt', async ({page}) => {
           await visit(page, {
-            id: 'components-forms-inlineautocomplete--default',
+            id: 'components-inlineautocomplete--default',
```
Comment (broccolinisoup) [inline e2e/components/InlineAutocomplete.test.ts:11]: Based on the draft status ([reference](https://github.com/primer/react/pull/3639#discussion_r1376911543)), we should be re-updating them again. Sorry I missed the component status in the first review.

Comment (broccolinisoup) [inline e2e/components/InlineAutocomplete.test.ts:11]: i.e.
```suggestion
            id: 'drafts-components-inlineautocomplete--default',
```

File: docs/content/drafts/InlineAutocomplete.mdx [hunk 4]
```diff
@@ -7,7 +7,7 @@ source: https://github.com/primer/react/tree/main/src/InlineAutocomplete
 storybook: '/react/storybook?path=/story/components-forms-inlineautocomplete--default'
```
Comment (broccolinisoup) [inline docs/content/drafts/InlineAutocomplete.mdx:7]:
```suggestion
storybook: '/react/storybook?path=/story/drafts-components-inlineautocomplete--default'
```

**Comment by green6erry** [comment]
@broccolinisoup, There are issues with naming it back to `Drafts/...` because of Tooptip v2. I'm not sure how you'd like to delineate between them, but I was just trying to get a story for the v1 Tooltip so there was something. I think it will be up to you or someone that's not me to decide.

File: e2e/components/InlineAutocomplete.test.ts [hunk 5]
```diff
@@ -64,7 +64,7 @@ test.describe('InlineAutocomplete', () => {
       test.describe(theme, () => {
         test('default @vrt', async ({page}) => {
           await visit(page, {
-            id: 'components-forms-inlineautocomplete-features--custom-rendering',
+            id: 'components-inlineautocomplete-features--custom-rendering',
```
Comment (joshblack) [inline e2e/components/InlineAutocomplete.test.ts:67]:
```suggestion
            id: 'drafts-components-inlineautocomplete-features--custom-rendering',
```

File: e2e/components/InlineAutocomplete.test.ts [hunk 6]
```diff
@@ -76,7 +76,7 @@ test.describe('InlineAutocomplete', () => {
 
         test('axe @aat', async ({page}) => {
           await visit(page, {
-            id: 'components-forms-inlineautocomplete-features--custom-rendering',
+            id: 'components-inlineautocomplete-features--custom-rendering',
```
Comment (joshblack) [inline e2e/components/InlineAutocomplete.test.ts:79]:
```suggestion
            id: 'drafts-components-inlineautocomplete-features--custom-rendering',
```

File: e2e/components/InlineAutocomplete.test.ts [hunk 7]
```diff
@@ -92,7 +92,7 @@ test.describe('InlineAutocomplete', () => {
       test.describe(theme, () => {
         test('default @vrt', async ({page}) => {
           await visit(page, {
-            id: 'components-forms-inlineautocomplete-features--single-line',
+            id: 'components-inlineautocomplete-features--single-line',
```
Comment (joshblack) [inline e2e/components/InlineAutocomplete.test.ts:95]:
```suggestion
            id: 'drafts-components-inlineautocomplete-features--single-line',
```

File: e2e/components/InlineAutocomplete.test.ts [hunk 8]
```diff
@@ -104,7 +104,7 @@ test.describe('InlineAutocomplete', () => {
 
         test('axe @aat', async ({page}) => {
           await visit(page, {
-            id: 'components-forms-inlineautocomplete-features--single-line',
+            id: 'components-inlineautocomplete-features--single-line',
```
Comment (joshblack) [inline e2e/components/InlineAutocomplete.test.ts:107]:
```suggestion
            id: 'drafts-components-inlineautocomplete-features--single-line',
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Looking great! ✨ Just left some comments on what I think the story ids ended up being with the move, hope they work 🤞

**Review by green6erry — COMMENTED** [review]
@broccolinisoup this also should be complete.

**Review by broccolinisoup — APPROVED** [review]

**Pull request merged**